packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -67,19 +68,32 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
 
   getChildren(element?: Tag): Thenable<TagTreeItem[]> {
     if (element) {
-      const references: TagReference[] = element.notes
+      const nestedTagItems: TagTreeItem[] = this.tags
+        .filter(item => item.tag.indexOf(element.tag + TAG_SEPARATOR) > -1)
+        .map(
+          item =>
+            new Tag(
+              item.tag,
+              item.tag.substring(item.tag.indexOf(TAG_SEPARATOR) + 1),
+              item.notes
+            )
+        )
+        .sort((a, b) => a.title.localeCompare(b.title));
+
+      const references: TagTreeItem[] = element.notes
         .map(({ uri }) => this.foam.workspace.get(uri))
-        .map(note => new TagReference(element.tag, note));
+        .map(note => new TagReference(element.tag, note))
+        .sort((a, b) => a.title.localeCompare(b.title));
 
       return Promise.resolve([
         new TagSearch(element.tag),
-        ...references.sort((a, b) => a.title.localeCompare(b.title)),
+        ...nestedTagItems.concat(references),
       ]);
     }
     if (!element) {
-      const tags: Tag[] = this.tags.map(
-        ({ tag, notes }) => new Tag(tag, notes)
-      );
+      const tags: Tag[] = this.tags
+        .filter(item => item.tag.indexOf(TAG_SEPARATOR) < 0)
```

Comment:
does this mean that tags that have children won't show up at top level?
if a workspace has the following tags: `#hello/world` and `#lalaland` wouldn't the first one be filtered here?
haven't tried it on my machine, so i might be missing something

Reply:
Interesting remark. I assumed a behaviour that would incline the parent tag always would be explicitly mentioned. Of course this is not ideal, so I changed the code to match the behaviour you described. Fixed in update of the PR.

Reply:
great, can you add an explicit test to check this behavior (in `foam-vscode`).
Also, can you add please add a test in the `markdown-provider.test.ts` to check that parent/children tags are picked up also from the front matter?

Reply:
@riccardoferretti I took a stab at tests. Needed to combine a bunch of things, so hopefully this is in line with the style being used.